runtime/lua/vim/lsp.lua

```diff
@@ -138,7 +138,7 @@ local function validate_command(input)
       end
     end
   else
-    error("cmd type must be string or list.")
+    error(string.format("cmd type must be string or list, but given %q.", type(input)))
```

Comment:
could remove this `else` block and use `vim.validate` at the beginning of `validate_command`. That will also ensure that the error message has a standard format.